### DISCUSSION TITLE
Attempting to implement a RAG solution. Having problems with ChatCompletionOptions

### DISCUSSION THREAD

**Discussion by rlevi-selly** [openai/openai-dotnet#304] [Q&A] [answered]
In Example03_FunctionCallingAsync.cs they show the following code. This is what I want to do, but the Tools property is read only.

```
        ChatCompletionOptions options = new()
        {
            Tools = { getCurrentLocationTool, getCurrentWeatherTool },
        };
```
How can I set my tools list correctly in a ChatCompletionOptions object in order to use them in a chat completion?

```
           var completionsResponse = await openAIClient.GetChatClient(configuration["OpenAI:Model"]).
                                                               CompleteChatAsync(messages: chatMessages, chatCompletionOptions)
```

Any help is appreciated!

**Comment by dspear** [comment, marked as answer]
It is a readonly collection, but you can Add to it or Remove from it. The init code effectively does an Add, which you can also do in code, for example:
options.Tools.Add(getCurrentLocationTool)

**Reply by rlevi-selly** [reply]
🤦🏻‍♂️duh...

Thanks!
